Source/DafnyCore/Coco/Parser.frame

```diff
@@ -223,12 +229,18 @@ public class Errors {
 
   // Code action range is up to, not including, the next token
   public static CodeActionToken CodeAction(IToken sourceTok, IToken next) {
-    return new CodeActionToken(sourceTok, sourceTok.col, next.pos - sourceTok.pos);
+    return CodeAction(sourceTok, sourceTok.col, next.pos - sourceTok.pos);
+  }
+
+  // Code action range is up to, not including, the given token's Next token 
+  // so the range includes following white space
+  public static CodeActionToken CodeActionToNext(IToken sourceTok) {
```

Comment:
Could you rename this just `CodeAction`? I think it's pretty intuitive that code actions on a single token will consider this token until the next.

Reply:
Whatever is the most frequent for single token should be just CodeAction(). The most frequent for removal is token to the next, and the most frequent for replacement is token to its final position.

I think what I am not comfortable with now is that we call this "CodeAction", whereas it has nothing to do with a code action, it's actually a *range* used by a code action.

So it might make sense to actually rename the class to CodeActionRange.

In any case, I think the case is perfect for an option, so I think for tokens the signature should be like this.

`CodeActionRange(IToken tok, bool IncludesTrailingSpace = false)`

By the way, now that I can think of it, you might want to only remove `token.TrailingTrivia` and not up to the next token. And even more, you might only want to remove the first consecutive whitespaces of `token.TrailingTrivia` that are not newlines. Here is why:

```
class X {
  static
  static /* comment about static */
  method Test() {
  }
}
```
In the example above, your code action would also remove the comment, not just the keyword. I think removing comments, in the unlikely event there are some, is not desired in any case.